Foundation/NSStream.swift

```diff
@@ -209,20 +227,32 @@ public class NSOutputStream : Stream {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        guard let property = property else { return false }
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
```

Comment:
I agree imho that this should not block feature development due to uncertain api discussion @parkera any thoughts on this? Can also can add tests for it too if its not compiled out via macro